### ISSUE TITLE
Problems with lr scheduler

### ISSUE THREAD

**Issue by xdedss** [ziqi-jin/finetune-anything#57]
https://github.com/ziqi-jin/finetune-anything/blob/1e5640d7796b078248b1c24a18ca34c6d834f025/extend_sam/__init__.py#L34C5-L34C18

The default args are not passed into get_scheduler in [train.py](https://github.com/ziqi-jin/finetune-anything/blob/1e5640d7796b078248b1c24a18ca34c6d834f025/train.py#L42)
```
        max_epoch=1,
        n_epochs_init=50,
        n_epochs_decay=50,
```

As a result, `cosine` and `linear` scheduler does not work properly. `cosine` will oscillate rapidly between 1 and 0 and `linear` will decrease the lr to negative values, causing the loss to explode

**Comment by yjdzyr** [comment]
Have you solved this problem? I have the same problem

**Comment by xdedss** [comment]
> Have you solved this problem? I have the same problem

a temporary fix is to remove [this line](https://github.com/ziqi-jin/finetune-anything/blob/1e5640d7796b078248b1c24a18ca34c6d834f025/train.py#L42) and add your own lr scheduler

**Comment by ayrus144** [comment]
> Have you solved this problem? I have the same problem

You can also set/ pass the max_epoch to be train_cfg.max_iter. Then the decay will happen gradually across the iteration steps.
